graphql-ballerina/Module.md

```diff
@@ -0,0 +1,685 @@
+                clientConfig: {
+                    secureSocket: {
+                       cert: {
+                           path: <truststore_path>,
+                           password: <password>
+                       }
+                    }
+                }
+            },
+            scopes: [<comma_separated_list_of_scopes>]
+        }
+    ]
+}
+service graphql:Service /graphql on new graphql:Listener(4000) {
+    // Service definition
+}
+```
+
+##### Maximum Query Depth
+When a maximum query depth is provided, all the queries exceeding that limit will be rejected at the validation phase and will not be executed.
+
+```ballerina
+import ballerina/graphql;
+
+@graphql:ServiceConfiguration {
+    maxQueryDepth: 2
+}
+service graphql:Service /graphql on new graphql:Listener(9090) {
+    // Service definition
+}
+```
+
+The above service only accepts queries of less than 2 levels. For an example, consider the following document:
+```
+query getData {
+    book {
+        author {
+            books {
+                author {
+                    books
+                }
+            }
+        }
+    }
+}
+```
+
+The result for the above query is the following JSON:
+
+```json
+{
+    "errors": [
+        {
+            "message": "Query \"getData\" has depth of 5, which exceeds max depth of 2",
+            "locations": [
+                {
+                    "line": 1,
+                    "column": 1
+                }
+            ]
+        }
+    ]
+}
+```
+
+### Types
+The Ballerina GraphQL resources can return the following types:
+
+#### Return Types
+
+##### Scalar types
+The following Ballerina types are considered as Scalar types:
+- `int`
+- `string`
+- `boolean`
+- `float`
+- `decimal`
+- `enum`
+
+```ballerina
+resource function get greeting() returns string {
+    return "Hello, World";
+}
+```
+
+This can be queried using the following document:
+```
+{
+    greeting
+}
+```
+
+Result:
+```json
+{
+    "data": {
+        "greeting": "Hello, World"
+    }
+}
+```
+
+##### Record types
+
+When a resource is returning a record type, each field of the record can be queried separately.
+Each record type is mapped to a GraphQL `OBJECT` type, and the field of the record type are mapped to the fields of the `OBJECT` type.
+
+```ballerina
+public type Person record {|
+    string name;
+    int age;
+|};
+
+resource function get profile() returns Person {
+    return { name: "Walter White", age: 51 };
+}
+```
+
+This will generate the following schema
+```
+type Query {
+    profile: Person!
+}
+
+type Person {
+    name: String!
+    age: Int!
+}
+```
+
+This can be queried using the following document:
+```
+{
+    profile {
+        name
+        age
+    }
+}
+```
+
+Result:
+```json
+{
+    "data": {
+        "profile": {
+            "name": "Walter White",
+            "age": 51
+        }
+    }
+}
+```
+
+Each field can be queried separately as shown in the following document:
```

Comment:
```suggestion
Each field can be queried separately as shown in the following document:
```
Is it correct to say "the following document"? Or, should it be "the following code"? Please check all occurrences.